examples/ImGuiModule.cmake

```diff
@@ -0,0 +1,184 @@
+include_guard()
+
+if(EXISTS "${ImGui_SRCDIR}" AND EXISTS "${ImGui_SRCDIR}/imgui.cpp")
+    set(ImGui TRUE)
+else()
+    include(${CMAKE_CURRENT_LIST_DIR}/ImGuiOptions.cmake)
+endif()
+
+function(imgui_option OPTION DESCRIPTION DEFAULT)
+    if(ImGui)
+        option(ImGui_${OPTION} "${DESCRIPTION}" ${DEFAULT})
+        set(ImGui_${OPTION}_DESCRIPTION "${DESCRIPTION}")
+        set(ImGui_${OPTION}_DEFAULT ${DEFAULT})
+        list(APPEND ImGui_OPTIONS ${OPTION})
+        set(ImGui_OPTIONS "${ImGui_OPTIONS}" PARENT_SCOPE)
+        set(ImGui_${OPTION}_DESCRIPTION "${DESCRIPTION}" PARENT_SCOPE)
+        set(ImGui_OPTIONS_CMAKE)
+        foreach(OPTION ${ImGui_OPTIONS})
+            string(APPEND ImGui_OPTIONS_CMAKE
+                "# ${ImGui_${OPTION}_DESCRIPTION}\nset(ImGui_${OPTION} ${ImGui_${OPTION}})\n")
+        endforeach()
+        set(ImGui_OPTIONS_CMAKE "${ImGui_OPTIONS_CMAKE}" PARENT_SCOPE)
+    endif()
+endfunction()
+
+function(imgui_export TARGET)
+    export(TARGETS ${TARGET} NAMESPACE ImGui::
+        FILE ImGui${TARGET}.cmake)
+    install(TARGETS ${TARGET} EXPORT ${TARGET})
+    install(EXPORT ${TARGET} NAMESPACE ImGui::
+        FILE ImGui${TARGET}.cmake
+        DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/imgui)
+endfunction()
+
+function(imgui_core)
+    set(TARGET Core)
```

Comment:
Why not just ImGui::ImGui? ImGui::Core doesn't sound as nice.

Reply:
This is agreed with the @ocornut and is his initiative.